### DISCUSSION TITLE
Add automated pull request comment that warns that a locale is "frozen"

### DISCUSSION THREAD

**Discussion by schalkneethling** [mdn/mdn-community#93] [Localization] [answered]
Quoting Peter from the original issue:

> If someone makes a PR on some `ru` or `pl` pages in mdn/translated-content, we should post a PR comment (automatically) that explains that this locale is not actively maintained (i.e. not unfrozen). 

https://github.com/mdn/content/issues/15381

**Comment by schalkneethling** [comment]
If we want to consider labeling as opposed to a comment, then this could be a boilerplate we can base this on:
https://github.com/mdn/content/blob/main/.github/workflows/issue-regex-labeler.yml

**Reply by schalkneethling** [reply]
Thinking about this more, I wonder how often we get pull requests for frozen locales. @mdn/localization-team-leads I would appreciate your input here. If we get very few, I will vote for either doing nothing or using the above labeling approach.

**Reply by teoli2003** [reply]
I have created some PRs against frozen locales to get rid of some macros, else we never can get rid of any of them (unless lucky).

**Reply by schalkneethling** [reply]
Thank you, @teoli2003. In these cases, though, I think adding a label would make more sense than informing the contributor that you are opening a pull request against a frozen locale that is not actively maintained. We can then use the label to create a view to know how many of these happen.

If we see a number of these being opened that grow stale over time, we might consider adding a comment instead of just a label.

**Reply by SphinxKnight** [reply]
FWIW, whenever new contributors create issues or PRs against such locales, I'm using a saved reply after applying a locale label (if necessary):
![image](https://user-images.githubusercontent.com/2413436/168727463-9f3f618f-9cf6-47e7-8f79-11f35c5b5951.png)
Here is the detailed answer: "Sorry, but this locale is currently frozen and changes/fixes are not currently accepted. Please see https://hacks.mozilla.org/2020/12/an-update-on-mdn-web-docs-localization-strategy/ and [README](https://github.com/mdn/translated-content/blob/main/README.md) of this repo."

Given the volume (and this tool (saved reply) I'm using), (and at least for me), this doesn't take too much time;

**Reply by schalkneethling** [reply]
@Rumyra ^^

**Comment by Rumyra** [comment]
The bigger question I think is what do we want to do with these frozen locales (I believe we have two German & Polish) they haven't been updated for 18 months now, so if anyone were to volunteer to work on them I think too much has changed for it to be effective. I'm happy to move them out of translated-content altogether, but am unsure of the repercussions of that from a platform perspective. Maybe @caugner or @fiji-flo has insight

**Reply by caugner** [reply]
If we decide to move de/pl out of translated-content, the required changes to yari are fairly easy and I don't expect any repercussions:

- Move the locales form `VALID_LOCALES` to `REDIRECT_LOCALES`.
- Remove translations from macros and a few other places (full-text search for `de`/`pl` on yari with the "Match whole word" option enabled)

That's all.

**Reply by schalkneethling** [reply]
Yeah, moving it is low impact on the platform. I am wondering, though, does that mean there is no path for those locales to ever get back on MDN Web Docs?

**Reply by SphinxKnight** [reply]
If I may comment here, I would second @Rumyra here and also (partly) reply to @schalkneethling regarding the current path for those locales. Without looking at a technical level in Yari, any German or Polish speaking team would have to:
- get their hands on Markdown conversion
- remove macros in a lot of pages (which may include `orphaned`/`conflicting` handling)

The tasks listed above are not, strictly speaking, localisation in themselves (e.g. any German page's content/wording would lightly benefit from it but not as much as a proper l10n update).

I don't know the state of German nor Polish at the moment, but should a new locale team want to contribute, I would consider offering them the choice to either (re)start from scratch (and work from `en-US` Markdown sources) vs from the current state of `files/de`/`files/pl`.

In terms of activity/"contribution signal", here are the list of:
- [`l10n-de` tagged issues and PRs](https://github.com/mdn/translated-content/issues?q=label%3Al10n-de) (~13 "organic" issues/PR were closed)
- [`l10n-pl` tagged issues and PRs](https://github.com/mdn/translated-content/issues?q=label%3Al10n-pl) (~7 "organic" issues/PR were closed)

**Reply by schalkneethling** [reply]
Thank you for the detailed feedback, @SphinxKnight. This helps a lot.

**Reply by schalkneethling** [reply]
I agree that at this point, starting fresh from `en-US` would be a better option for sure.

**Reply by Rumyra** [reply]
Yeh I've been pretty much thinking the same as @SphinxKnight - if these locales do want to unfreeze in the future, they would have to do the markdown conversion and macro removal _first_ before they are moved back - add to that all the updates that would be needed now, there's a strong case to start again (although that's also a lot of work).

**Reply by SphinxKnight** [reply]
(for cross-reference: https://github.com/mdn/mdn-community/discussions/147 )

**Comment by caugner** [comment, marked as answer]
We have retired German and Polish, so this is no longer an issue. 🎉
